### ISSUE TITLE
e-mail list column is too narrow, bulky and generally lacks in visual clarity

### ISSUE THREAD

**Issue by cptMikky** [nextcloud/mail#3940]
#### Summary

As a user don't like the current layout in terms of font size, text margins and column width.

The folder list is unnecessarily wide while the e-mail list is too narrow and bulky, hiding too much information needlessly (even not-too-long subject lines are truncated) and squeezing a lot of information into a small space.

The e-mail content column, on the other hand, is so wide that the side margins are about 1/3 of the content width of unused space each (on a common desktop widescreen display) which gives more than enough space to extend the e-mail list column without sacrificing anything.

The "bulkiness" feel of the e-mail list column is, for me, caused mainly by these factors:

1. the column is too narrow which causes information loss and compression (see above)
2. the font of both Sender and Subject is unnecessarily large
3. there is no visual clue to separate the individual e-mails 
4. the distinction between read and unread e-mails is not visually strong enough
5. the attachment icon makes the list look visually inconsistent

#### Preferred solution

I suggest these fixes respective to the above mentioned factors:

1. allow manual resize of the column(s) (or make the e-mail list column about twice as wide)
2. decrease font size by 1-2 points and make the subject font color darker (or black, same as sender)
3. add a thin light line between the entries. This and 2. may allow for slightly tighter margins making the list more compact.
4. make unread e-mail background color slightly darker or a different color
5. have a fixed place for the paperclip icon that doesn't change visual arrangement of the text

**Comment by ChristophWurst** [comment]
@jancborchardt 👀

**Comment by dumblob** [comment]
I considered this a low-priority, so I didn't want to clutter this bug tracker. But it appeared here, so I feel obliged to raise my hand as I have exactly the same issues as @cptMikky . Making it less "bulkier" in one way or another would be very helpful for everyday users.

**Comment by brueckner** [comment]
I fully agree! **However**, I believe this is one of those things that heavily depend on the user's taste and use-case. Personally, I couldn't care less about avatars in an email client to be honest, but it might be useful for others to visually scan for the email they're looking for. 

What I am trying to say is: there is no "one size fits all" solution here, I think.

Instead, I think this is something that needs to be **configurable**, just like in other email clients. Take Apple Mail for example:

<img width="548" src="https://user-images.githubusercontent.com/459512/98093903-d867d100-1e88-11eb-9ddb-9f907f1ec256.png">

I think it is reasonable to have "shiny" default settings (that is: spacious, avatars, etc.) with "power user" settings to make things more compact.

**Comment by cptMikky** [comment]
Agree with Johannes - a bunch of configuration options would be much better than changing one hardcoded layout to another. 

However, it is my current understanding that the preferred "nextcloud way" is to have the UI more standardized for some reason so my suggestion was more along that line. I'd be happy either way but I _think_ at least power users (yours truly included) would be happier with more granular UI configuration.

Same for #3941

**Comment by ChristophWurst** [comment]
> Agree with Johannes - a bunch of configuration options would be much better than changing one hardcoded layout to another.

Sarkasm doesn't help the conversation.

**Comment by cptMikky** [comment]
> Sarkasm doesn't help the conversation.

?

**Comment by ChristophWurst** [comment]
Was it not meant to be sarcastic? Then I'm really sorry. I read the *bunch of configuration options* wrong. My bad.

**Comment by cptMikky** [comment]
Not in the slightest :) The comment was meant to say that I'd be happier with having even a higher amount of config knobs and fields over any hardcoded setup. I was just trying to stress that it is merely my _personal_ view backed by nothing more than my own experience and wishes and, to some extend, wishes of my users (who mostly described the desired state rather than the way to achieve it).

**Comment by ChristophWurst** [comment]
Got it :)

**Comment by ei8fdb** [comment]
(I came to open the exact same ticket and was happy to find it was already open!)

I agree with the ticket subject "list column is too narrow, and generally lacks in visual clarity". I also agree with @brueckner's comment how a user wants to display email is part personal preference.

My preference would be to support common mailbox layouts:

- classic
- wide view
- vertical

I'd like to also mention being able to resize the mail column view, also a common control in mail clients.

**Comment by alexanderdd** [comment]
-> Use (up to) two lines for the subject.

What do you think? And did anyone already suggest this, couldn't find it via search?

@column width: that is probably hardcoded for all apps?! And should follow a similar logic in all apps!?

**Comment by politas** [comment]
What I would really like to see is an option to have the list and preview columns above-and-below rather than side-by-side. And I think adjustable layout is absolutely required.

**Comment by TheNomad11** [comment]
Totally agree. Now it is hardly usable which is a pity as Mail is such a useful NC app. But I suppose it should be possible by editing the css somewhere somehow? Or by custom theming? Shouldn't be a big issue, I suppose?

**Comment by ChristophWurst** [comment]
There is some progress at https://github.com/nextcloud/mail/pull/4123

**Comment by TheNomad11** [comment]
Cool, that's good news. Thanks a lot!

**Comment by bananacy** [comment]
Install the Custom CSS APP then under Theming > custom css ad this:

.app-content-list {   
    max-width: 400px !important;
}

Or any other css modifications :-)

**Comment by ei8fdb** [comment]
> Install the Custom CSS APP then under Theming > custom css ad this:
> 
> .app-content-list {
> max-width: 400px !important;
> }
> 
> Or any other css modifications :-)

Heh, I know that's 50% a joke, but I don't think it's a usable solution for about 90% of people. Though, thank you for the tip, I didn't know if existed. I'll see if I can use it without making a mistake!

**Comment by alexanderdd** [comment]
I don't understand why it's 50% joke? It works very well for us, I even set the width to 500px. When you make the window smaller, the list gets smaller automatically.

Can someone turn this CSS snippet into a pull request? I feel 400px is a much more sensible default than now. Resizing https://github.com/nextcloud/mail/pull/4123 is also good, but we need a sensible default anyway.

**Comment by dumblob** [comment]
Speaking of changing the default - shouldn't the default be rather font-size dependent than pixel-size dependent? I mean why not `30rem` instead of `400px`? `rem` is what works best for tablets & desktops in my case (unlike `px` which is often wrong).

**Comment by ChristophWurst** [comment]
https://github.com/nextcloud/mail/pull/5201 will be shipped with v1.10 in a few days, I think this ticket is obsolete then.

**Comment by alexanderdd** [comment]
Does https://github.com/nextcloud/mail/pull/5201 also change default column width?

If it doesn't, then I think this issue should stay open. Resizing is not enough.

**Comment by BEEDELLROKEJULIANLOCKHART** [comment]
@alexanderdd, I have reported that problem separately at "http://github.com/nextcloud/mail/issues/6525".
